### ISSUE TITLE
Fastify support

### ISSUE THREAD

**Issue by wyozi** [inngest/inngest-js#249]
Would be great if you supported fastify (https://www.fastify.io/) similarly to how express support is implemented

Seems like #202 would fix this

**Comment by djfarrelly** [comment]
Hi @wyozi, to better understand how Fastify users might best want to use Inngest, do you think it would be better as a Fastify plugin (e.g. via `fastify.register()`) or a serve handler passed to `fastify.route()`?

### Plugin concept:
```ts
const inngest = new Inngest({ name: "My app" });
const myFunction = inngest.createFuncton(...);

import { inngestFastify } from "@inngest/fastify"; // new package

await fastify.register(inngestFastify, { // New plugin idea
  url: "/api/inngest",
  client: inngest,
  functions: [myFunction],
}
```

### Route handler

This is most similar to #202:

```ts
const inngest = new Inngest({ name: "My app" });
const myFunction = inngest.createFuncton(...);

import { serve } from "inngest/fastify"; // see #202 

 await fastify.route({
    method: ['GET', 'POST', 'PUT'],
    url: '/api/inngest',
    handler: serve(inngest, [myFunction]),
  })
```

---

Both options are possible, but I don't know enough about Fastify to know when something should be a plugin vs. just a route handler. I appreciate any insight!

**Comment by wyozi** [comment]
@djfarrelly good question!

I'd say route handler only might make the most sense, as the way different projects use plugins tends to vary. Some may want to have more control over the plugin hierarchy, which would be served best by route only imo. Also looked at how Yoga does it, and they do it the route handler way (https://the-guild.dev/graphql/yoga-server/docs/integrations/integration-with-fastify)

**Comment by djfarrelly** [comment]
Thanks for the input @wyozi! I did also get to connect w/ Matteo who is the lead maintainer of Fastify and he recommended the Plugin option, something like this:

```ts
import { Inngest } from "inngest"
import { inngestFastify } from "@inngest/fastify"

const inngest = new Inngest({ name: "Acme API" });
const sendWelcomeEmailFn = inngest.createFunction({ name: "Send welcome email" }, { event: "user.signup" }, ({ event }) => {/* function logic */})

await fastify.register(inngestFastify, {
  url: "/api/inngest",
  client: inngest,
  functions: [sendWelcomeEmailFn],
}
```

I think this would still allow someone to configure the route, although I'd have to understand more about what you mean around plugin hierarchy in terms of the benefits of a route.

**Comment by wyozi** [comment]
@djfarrelly I was more thinking in terms of plugin encapsulation and how to maximize control over route declaration, but a self-contained plugin that allows passing in all required configuration would check all the same boxes. And especially if Matteo thinks that's the best method, I would trust him on that :)

**Comment by jpwilliams** [comment]
@wyozi Fastify support has been released in [v2.6.0](https://github.com/inngest/inngest-js/releases/tag/inngest%402.6.0) following #202. 🙂 

You can see how to use the new handler over at [Serving the Inngest API - Framework: Fastify](https://www.inngest.com/docs/sdk/serve#framework-fastify) - it's basically the same as has been discussed here.

```ts
import Fastify from "fastify";
import inngestFastify from "inngest/fastify";
import { inngest, fnA } from "./inngest";

const fastify = Fastify();

fastify.register(inngestFastify, {
  client: inngest,
  functions: [fnA],
  options: {},
});

fastify.listen({ port: 3000 }, function (err, address) {
  if (err) {
    fastify.log.error(err);
    process.exit(1);
  }
});
```

Let us know if you have any problems!
